src/core/diags/factory_diags.cpp

```diff
@@ -316,12 +340,25 @@ Error Diags::ProcessSend(uint8_t aArgsLength, char *aArgs[], char *aOutput, size
     long  value;
 
     VerifyOrExit(otPlatDiagModeGet(), error = kErrorInvalidState);
-    VerifyOrExit(aArgsLength == 2, error = kErrorInvalidArgs);
+    VerifyOrExit(aArgsLength >= 1, error = kErrorInvalidArgs);
 
     SuccessOrExit(error = ParseLong(aArgs[0], value));
     mTxPackets = static_cast<uint32_t>(value);
 
-    SuccessOrExit(error = ParseLong(aArgs[1], value));
+    if (aArgsLength >= 2)
+    {
+        SuccessOrExit(ParseLong(aArgs[1], value));
+        mTxPacketReady = false;
+    }
+    else if (mTxPacketReady)
+    {
+        value = mTxPacket->mLength;
+    }
+    else
+    {
+        ExitNow(error = kErrorInvalidArgs);
```

Comment:
I think we can set a default length of the frame here and update the README to notify users.

Reply:
It's not the aim of this PR. Someone can make the change when it's needed. Thoughts?